src/app/App.js

```diff
@@ -612,7 +619,7 @@ class App extends EventEmitter {
 		});
 		if (!utils.isCurrentBrowserPath(path)) {
 			if (this.isNavigationPending && this.pendingNavigate) {
-				this.pendingNavigate.thenAlways(() => this.removeScreen(path), this);
+				this.pendingNavigate.then(() => this.removeScreen(path), this);
```

Comment:
To preserve the existing behavior should this be?

```javascript
this.pendingNavigate
  .then(() => this.removeScreen(path))
  .catch(error => {
    this.removeScreen(path);
    throw error;
  });
```

I haven't seen the definition of `thenAlways()` yet, but I am assuming the second `this` parameter is to set the context of the callback function? If so, it was a no-op in the old function (because we were using an arrow), and in the new version we shouldn't be passing `this` as a second param to `then()` because that is supposed to be an `onReject` handler.

Reply:
Yes you're right, good catch.